src/geode/geosciences/lso_input.cpp

```diff
@@ -0,0 +1,783 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/geosciences/private/lso_input.h>
+
+#include <algorithm>
+#include <fstream>
+#include <functional>
+
+#include <absl/container/flat_hash_set.h>
+#include <absl/strings/numbers.h>
+#include <absl/strings/str_split.h>
+
+#include <geode/basic/attribute_manager.h>
+
+#include <geode/geometry/bounding_box.h>
+#include <geode/geometry/nn_search.h>
+#include <geode/geometry/point.h>
+#include <geode/geometry/vector.h>
+
+#include <geode/mesh/builder/edged_curve_builder.h>
+#include <geode/mesh/builder/point_set_builder.h>
+#include <geode/mesh/builder/tetrahedral_solid_builder.h>
+#include <geode/mesh/builder/triangulated_surface_builder.h>
+#include <geode/mesh/core/edged_curve.h>
+#include <geode/mesh/core/geode_triangulated_surface.h>
+#include <geode/mesh/core/mesh_factory.h>
+#include <geode/mesh/core/tetrahedral_solid.h>
+
+#include <geode/model/mixin/core/block.h>
+#include <geode/model/mixin/core/corner.h>
+#include <geode/model/mixin/core/line.h>
+#include <geode/model/mixin/core/surface.h>
+#include <geode/model/mixin/core/vertex_identifier.h>
+#include <geode/model/representation/io/brep_output.h>
+
+#include <geode/geosciences/private/gocad_common.h>
+#include <geode/geosciences/representation/builder/structural_model_builder.h>
+#include <geode/geosciences/representation/core/structural_model.h>
+
+namespace
+{
+    class LSOInputImpl
+    {
+    public:
+        static constexpr geode::index_t OFFSET_START{ 1 };
+        static constexpr char EOL{ '\n' };
+
+        LSOInputImpl(
+            absl::string_view filename, geode::StructuralModel& model )
+            : file_( filename.data() ),
+              model_( model ),
+              builder_( model ),
+              solid_{ geode::TetrahedralSolid3D::create() },
+              solid_builder_{ geode::TetrahedralSolidBuilder3D::create(
+                  *solid_ ) },
+              vertex_id_{
+                  solid_->vertex_attribute_manager()
+                      .find_or_create_attribute< geode::VariableAttribute,
+                          geode::index_t >( "vertex_id", geode::NO_ID )
+              }
+        {
+            OPENGEODE_EXCEPTION( file_.good(),
+                "[LSOInput] Error while opening file: ", filename );
+        }
+
+        void read_file()
+        {
+            geode::detail::check_keyword( file_, "GOCAD LightTSolid" );
+            geode::detail::read_header( file_ );
+            crs_ = geode::detail::read_CRS( file_ );
+            read_vertices();
+            read_tetra();
+            read_surfaces();
+            read_blocks();
+            build_model_boundaries();
+            build_corners();
+            build_lines();
+        }
```

Comment:
Add a message that tell the user that properties are not supported. neither on vertex nor on cells